torch/linalg/__init__.py

```diff
@@ -1480,6 +1480,22 @@
   Please use :func:`torch.linalg.svdvals`, which computes only the singular values,
   instead of `compute_uv=False`.
 
+.. note:: Selection of `driver` (this keyword argument only works on CUDA inputs with cuSOLVER backend):
+
+    - When `driver` is not given or set to `None` (*default*), a heuristic of drivers will be selected based on
+      matrix sizes and batch sizes of the inputs.
+      The default choice should be efficient and precise for most inputs. However, if input matrices
+      have large condition numbers or could be ill-conditioned, it's recommended to choose the `gesvd` driver.
+    - The `gesvd` driver directly calls the cuSOLVER `cusolverDn<t>gesvd` method. This QR-based algorithm is
+      less performant on GPU, but it can converge ill-conditioned matrices and get high precision for those inputs.
+    - The `gesvdj` driver directly calls the cuSOLVER `cusolverDn<t>gesvdj` method. This Jacobi-based algorithm
+      has better performance than `gesvd` on GPU, but it may get less accurate results for ill-conditioned matrices or
+      matrices with very large sizes (be cautious if matrix size > 2048).
+    - The `gesvdjBatched` driver directly calls the cuSOLVER `cusolverDn<t>gesvdjBatched` method. It only works for
```

Comment:
Remind me: were we thinking of the "gesvdj" calling either gesvdj or gesvdjBatched for the user?

Reply:
The discussion was to merge `gesvdj` and `gesvdjBatched` to one option, and select the actual driver in the backend based on input shapes.

Now I think it's a better idea to merge these two methods. With that, each of the three drivers would work for (nearly) all shapes.